prow/bugzilla/client.go

```diff
@@ -172,30 +182,37 @@ func getRecursiveClones(c Client, root *Bug) ([]*Bug, error) {
 }
 
 // getImmediateParents gets the Immediate parents of bugs with a matching summary
-func getImmediateParents(c Client, bug *Bug) ([]*Bug, error) {
+func getImmediateParents(c Client, bug *Bug, cache map[int]*Bug) ([]*Bug, error) {
 	var errs []error
 	parents := []*Bug{}
 	// One option would be to return as soon as the first parent is found
 	// ideally that should be enough, although there is a check in the getRootForClone function to verify this
 	// Logs would need to be monitored to verify this behavior
 	for _, parentID := range bug.DependsOn {
-		parent, err := c.GetBug(parentID)
-		if err != nil {
-			errs = append(errs, fmt.Errorf("Failed to get parent bug #%d: %v", parentID, err))
-			continue
+
+		parent, ok := cache[parentID]
```

Comment:
```
parent, hasParent := cache[parentID]
if hasParent {
   continue
}
/// GetParent
```

Reply:
I don't think I can do that unless I duplicate the code which checks if it is a clone - which needs to be done both in case its present in the cache or not. (Unless I'm missing something here)